workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs

```diff
@@ -0,0 +1,192 @@
+
+        private string requiredWorkerTypesLabel;
+        private string workerType;
+        private bool? isWorkerType;
+
+        private bool foldout;
+
+        private GUIContent enabledContent;
+        private GUIContent disabledContent;
+
+#if ODIN_INSPECTOR
+        protected override void OnEnable()
+#else
+        protected void OnEnable()
+#endif
+        {
+            // Fetch styles
+            enabledContent = EditorGUIUtility.IconContent("sv_icon_dot3_pix16_gizmo");
+            disabledContent = EditorGUIUtility.IconContent("sv_icon_dot0_pix16_gizmo");
+
+            // Get type info
+            script = (MonoBehaviour) target;
+            scriptType = script.GetType();
+            isSpatialBehaviour = HasWorkerTypeAttribute(scriptType) || HasRequireAttributes(scriptType);
+            if (!isSpatialBehaviour)
+            {
+                return;
+            }
+
+            linkedEntityComponent = script.GetComponent<LinkedEntityComponent>();
+            var requiredWorkerTypes = GetRequiredWorkerTypes(scriptType);
+
+            subscriptions = new Dictionary<FieldInfo, ISubscription>();
+            var subscriptionsInfo = RequiredSubscriptionsDatabase.GetOrCreateRequiredSubscriptionsInfo(scriptType);
+
+            // Get subscription info when playing
+            if (Application.isPlaying && linkedEntityComponent != null)
+            {
+                workerType = linkedEntityComponent.Worker.WorkerType;
+                isWorkerType = requiredWorkerTypes?.Contains(workerType);
+                if (isWorkerType.HasValue)
+                {
+                    requiredWorkerTypesLabel = string.Join(" || ", requiredWorkerTypes);
+                }
+
+                var subscriptionSystem = linkedEntityComponent.World.GetExistingSystem<SubscriptionSystem>();
+
+                foreach (var fieldInfo in subscriptionsInfo.RequiredFields)
+                {
+                    var subscription =
+                        subscriptionSystem.Subscribe(linkedEntityComponent.EntityId, fieldInfo.FieldType);
+                    subscriptions.Add(fieldInfo, subscription);
+                }
+            }
+            else
+            {
+                if (requiredWorkerTypes != null)
+                {
+                    requiredWorkerTypesLabel = string.Join(" || ", requiredWorkerTypes);
+                }
+
+                foreach (var fieldInfo in subscriptionsInfo.RequiredFields)
+                {
+                    subscriptions.Add(fieldInfo, null);
+                }
+            }
+        }
+
+#if ODIN_INSPECTOR
+        protected override void OnDisable()
+#else
+        protected void OnDisable()
+#endif
+        {
+            if (subscriptions != null)
+            {
+                foreach (var pair in subscriptions)
+                {
+                    var subscription = pair.Value;
+                    subscription?.Cancel();
```

Comment:
Under what conditions should the subscription be null here?

Reply:
In Editor mode the subscriptions are null

Reply:
Ah right